pkg/controllers/restore/controller.go

```diff
@@ -477,6 +476,26 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 	return nil
 }
 
+// customize provides customization of restored resource for edge cases
+func customize(obj *unstructured.Unstructured) {
+	switch obj.GetKind() {
+	case "ServiceAccount":
+		// remove secrets section as referenced secrets will be removed by k8s Token Controller as they are considered orphaned
+		delete(obj.Object, secretsMapKey)
+		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
+	case "Cluster":
+		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
+		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+			n, err := rand.Int(rand.Reader, big.NewInt(99999))
+			if err == nil {
+				if err := unstructured.SetNestedField(obj.Object, n.Int64(), "spec", "redeployAgentGeneration"); err != nil {
```

Comment:
What exactly is `redeployAgentGeneration` and why is it acceptable to set random value to it? Assuming it's already set to some value, what if random happens to return the same value as well?

Reply:
It will redeploy agent in downstream cluster which will reissue serviceaccount token. Yeah will be good if we check if value matches and regenerate it.